source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,177 @@
+
+```nix
+{ config, pkgs, ... }:
+{
+  imports =  [ ./hardware-configuration.nix ];
+
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  system.stateVersion = "22.05";
+}
+```
+
+In this tutorial we focus on testing NixOS configurations on a virtual machine.
+Therefore we will remove the reference to `hardware-configuration.nix`:
+
+```diff
+-  imports =  [ ./hardware-configuration.nix ];
+```
+
+Changes to the configuration need to be positioned inside the curly bracket.[^bracket]
+
+[^bracket]: As a reminder `configuration.nix` contains a function that returns an [attribute set](https://nixos.org/manual/nix/stable/language/values.html#attribute-set) that follows the convention of a [module](https://nixos.org/manual/nixos/stable/index.html#sec-writing-modules).
+In the attribute set you describe how you want your NixOS system configured.
+
+To be able to log in add the following lines:[^user]
+
+[^user]: On NixOS your configuration generated using `nix-generate-config` contains the user configuration commented out.
+
+```nix
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ];
+    packages = with pkgs; [
+      firefox
+      thunderbird
+    ];
+  };
+```
+
+Additionally, you need to specify a password for this user.
+For the purpose of demonstration only, we use specify an insecure, plain text password by adding the `initialPassword` option to the user configuration:[^password]
+
+[^password]: Warning: Do not use plain text passwords outside of this example unless you know what you are doing. See [`initialHashedPassword`] or [`ssh.authorizedKeys`] for more secure alternatives.
+
+```nix
+initialPassword = "testpw";
+```
+
+The complete `configuration.nix` file now looks like this:
+
+```nix
+{ config, pkgs, ... }:
+{
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ]; # Enable ‘sudo’ for the user.
+    packages = with pkgs; [
+      firefox
+      thunderbird
+    ];
+    initialPassword = "testpw";
+  };
+
+  system.stateVersion = "22.05";
+}
+```
+
+## Creating a QEMU based virtual machine using a configuration
+
+A virtual machine is created with the `nix-build` command.
+
+To select `configuration.nix` in the working directory, specify the configuration file as an argument:[^nixosrebuild]
+
+[^nixosrebuild]: On NixOS you can create a virtual machine using `nixos-rebuild build-vm -I nixos-config=./configuration.nix`, which wraps the original command.
+
+```shell-session
+nix-build '<nixpkgs/nixos>' -A vm -I nixos-config=./configuration.nix
+```
```

Comment:
Since [you asked on Discourse](https://discourse.nixos.org/t/where-is-a-vm-documented/22258/2) about this being documented, we may want to add some explanation in case someone wants to know more. It's irrelevant for continuing the tutorial, so it should be an optional read.

````suggestion
```shell-session
nix-build '<nixpkgs/nixos>' -A vm -I nixos-config=./configuration.nix
```
<details><summary>Detailed explanation</summary>

TODO: explain `<nixpkgs/nixos>`

arguments of nix-build: https://nixos.org/manual/nix/stable/command-ref/nix-build.html
search paths: https://nixos.org/manual/nix/stable/command-ref/env-common.html#env-NIX_PATH

TODO: explain what `-A` does (https://nixos.org/manual/nix/stable/command-ref/opt-common.html#opt-attr)

TODO: explain where `vm` comes from (https://github.com/NixOS/nixpkgs/blob/7c164f4bea71d74d98780ab7be4f9105630a2eba/nixos/default.nix#L19)

TODO:  explain what `-I` does here (https://nixos.org/manual/nix/stable/command-ref/opt-common.html#opt-I)

</details>
````